### PR TITLE
translate platform environment

### PR DESCRIPTION
**Pull request by jessie-zhao**
translating platform environment

### CODE REVIEW

**Comment by kuri-leo** [comment]
Mark, do it tomorrow

File: site/zh-cn/js/guide/platform_environment.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+在Node.js平台上，TensorFlow.js既支持直接使用TensorFlow API，也支持更慢的CPU环境。
+
+## [](https://github.com/tensorflow/tfjs-website/blob/master/docs/guide/platform_environment.md#environments)环境
+
+当一个用TensorFlow.js开发的程序运行时，所有的配置被统称为环境。它包含一个全局的backend，以及一些可以精确控制TensorFlow.js特性的标记。
+
+### [](https://github.com/tensorflow/tfjs-website/blob/master/docs/guide/platform_environment.md#backends)Backends
+
+TensorFlow.js支持多个不同的Backend，用来实现存储和数学操作。任何时候都只有一个backend生效。大部分时间，TensorFlow.js会根据当前环境自动选择使用最佳的backend。即使这样，你仍然需要知道，如何得知当前正在使用的是哪个backend，以及如何在不同backend之间切换。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/platform_environment.md]:
```suggestion
TensorFlow.js支持多个不同的backend，用来实现张量的存储和数学操作。任何时候都只有一个backend生效。大部分时间，TensorFlow.js会根据当前环境自动选择使用最佳的backend。即使这样，你仍然需要知道，如何得知当前正在使用的是哪个backend，以及如何在不同backend之间切换。
```
补充

File: site/zh-cn/js/guide/platform_environment.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+```
+
+#### [](https://github.com/tensorflow/tfjs-website/blob/master/docs/guide/platform_environment.md#webgl-backend)WebGL backend
+
+WebGL backend，简称“webgl”，是在浏览器平台上最强大的一个backend。它比CPU backend要快100倍。部分原因是，Tensor是作为WebGL纹理保存的，数学运算操作实现在WebGL shader里面。
+
+下面是在使用这个backend时需要了解的一些知识。
+
+##### [](https://github.com/tensorflow/tfjs-website/blob/master/docs/guide/platform_environment.md#avoid-blocking-the-ui-thread)避免阻塞UI线程
+当调用一个操作，如tf.matMul(a,b)时，返回值tf.Tensor会同步返回，然而这时矩阵乘法运算还不一定完成。这意味着返回值tf.Tensor只是一个指向运算的句柄。当调用x.data()或x.array()时，只有当运算完成时才能取到实际值。因此在运算过程中，为避免阻塞UI线程，需要使用异步版本的x.data()和x.array()，而不是同步版本的x.dataSync()和x.arraySync()。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/platform_environment.md]:
```suggestion
当调用一个操作，如tf.matMul(a,b)时，返回值tf.Tensor会同步返回，然而这时矩阵乘法运算还不一定完成。这意味着返回值tf.Tensor只是一个指向运算的句柄。当调用`x.data()`或`x.array()`时，只有当运算完成时才能取到实际值。因此在运算过程中，为避免阻塞UI线程，需要使用异步版本的`x.data()`和`x.array()`，而不是同步版本的`x.dataSync()`和`x.arraySync()`。
```
根据原文补充markdown代码格式

File: site/zh-cn/js/guide/platform_environment.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+##### [](https://github.com/tensorflow/tfjs-website/blob/master/docs/guide/platform_environment.md#shader-compilation--texture-uploads)编译Shader& texture 上传
+TensorFlow.js在GPU里执行WebGL的shader程序。然而这些shader只有在被调用时才会被编译，即lazy-compile。编译过程在CPU上的主线程完成，这导致程序变慢。TensorFlow.js会自动缓存编译好的shader，下次再调用有同样shape，同样输入输出的tensor时能快很多。TensorFlow.js开发的应用一般会多次使用同样的操作，因此第二次运行会快很多。
+
+TensorFlow.js还会把tf.Tensor数据存储为WebGL纹理。当一个tf.Tensor被创建后，不会被立即上传到GPU，而是当其被用到时才这么做。如果这个tf.Tensor被第二次使用，由于已经在GPU里，因此省掉了上传开销。在一个典型的机器学习模型中，这意味着weight在第一次预测时被上传，第二次就会快很多。
+
+如果希望加快第一次预测的性能，我们推荐对模型进行预热，即传递一个有同样shape的输入Tensor。
+例如:
+```js
+const model = await tf.loadLayersModel(modelUrl);
+// Warmup the model before using real data.
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/platform_environment.md]:
```suggestion
// 使用真实数据来预热模型
```

File: site/zh-cn/js/guide/platform_environment.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+const warmupResult = model.predict(tf.zeros(inputShape));
+warmupResult.dataSync();
+warmupResult.dispose();
+
+// The second predict() will be much faster
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/platform_environment.md]:
```suggestion
// 第二次执行 predict() 的时候将会更加快速
```

File: site/zh-cn/js/guide/platform_environment.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+const result = model.predict(userData);
+```
+
+#### [](https://github.com/tensorflow/tfjs-website/blob/master/docs/guide/platform_environment.md#nodejs-tensorflow-backend)Node.js TensorFlow backend
+
+在Node.js TensorFlow backend中，TensorFlow的C语言API被用来加速操作。它会尽可能使用机器的硬件加速模块，如CUDA。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/platform_environment.md]:
```suggestion
在Node.js TensorFlow backend中，所谓“node”,即TensorFlow的C语言API被用来加速操作。它会尽可能使用机器的硬件加速模块，如CUDA。
```

File: site/zh-cn/js/guide/platform_environment.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+在这个backend中，和WebGL backend一样，函数会同步返回tf.Tensor。然而，与WebGL backend不同的是，当你获得这个tensor返回值时，运算已经完成。这意味着tf.matMul(a,b)调用会阻塞UI线程。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/platform_environment.md]:
```suggestion
在这个backend中，和WebGL backend一样，函数会同步返回`tf.Tensor`。然而，与WebGL backend不同的是，当你获得这个tensor返回值时，运算已经完成。这意味着`tf.matMul(a,b)`调用会阻塞UI线程。
```

File: site/zh-cn/js/guide/platform_environment.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+这个backend是性能最差的backend，然而是最简单的。所有操作都实现在vanilla JavaScript中，因此很少有并行化，并且会阻塞UI线程。
+
+这个backend对测试有用，或者是用于WebGL不能使用的设备。
+
+### [](https://github.com/tensorflow/tfjs-website/blob/master/docs/guide/platform_environment.md#flags)Flags
+
+TensorFlow.js有一套环境标记，能够自动评估和检测，保证是当前平台上的最佳配置。这些标记大部分是内部使用，其中有一些全局标记可以被API控制。
+
+-   `tf.enableProdMode():`  允许生产模式。它会去掉模型验证，NaN检查，以及其他错误校验操作，从而提高性能。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/platform_environment.md]:
```suggestion
-   `tf.enableProdMode():`  启用生产模式。它会去掉模型验证，NaN检查，以及其他错误校验操作，从而提高性能。
```

File: site/zh-cn/js/guide/platform_environment.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+-   `tf.enableDebugMode()`: 允许调试模式。它会记录每种操作的日志并输出到到调试台，还记录运行性能信息，如内存footprint和内核执行时间。注意这将极大降低应用运行时间，不可在生产环境中使用。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/platform_environment.md]:
```suggestion
-   `tf.enableDebugMode()`: 启用调试模式。它会记录每种操作的日志并输出到到调试台，还记录运行性能信息，如内存footprint和内核执行时间。注意这将极大降低应用运行时间，不可在生产环境中使用。
```

**Comment by lamberta** [comment]
Hi jessie-zhao@, I've confirmed you signed the CLA but these commits are using a different email address than the one you signed the CLA with.
Can you also sign with this email? https://cla.developers.google.com/
Thanks

**Comment by kuri-leo** [comment]
Hi @lamberta , it seems like my review suggestions bring those conflicts (and for many times :sob: . But I did signed the CLA. is this a bug?

**Comment by lamberta** [comment]
@kuri-leo It's not you :)
It looks like @jessie-zhao signed the CLA with his personal email but submitted these commits with his work email. Not a problem, but should sign the CLA again to avoid confusion in the future.

**Comment by kuri-leo** [comment]
Ok, I will tell her tomorrow ASAP (it's midnight in China
thank you @lamberta

**Pull request merged**